dbt-snowflake/src/dbt/include/snowflake/macros/relations/dynamic_table/create.sql

```diff
@@ -62,16 +64,18 @@ create dynamic table {{ relation }}
 
 {%- set catalog_relation = adapter.build_catalog_relation(config.model) -%}
 
-create dynamic iceberg table {{ relation }}
-    target_lag = '{{ dynamic_table.target_lag }}'
-    warehouse = {{ dynamic_table.snowflake_warehouse }}
-    {{ optional('external_volume', catalog_relation.external_volume, "'") }}
-    catalog = 'SNOWFLAKE'  -- required, and always SNOWFLAKE for built-in Iceberg tables
-    base_location = '{{ catalog_relation.base_location }}'
-    {{ optional('refresh_mode', dynamic_table.refresh_mode) }}
-    {{ optional('initialize', dynamic_table.initialize) }}
-    as (
-        {{ sql }}
-    )
+    create dynamic iceberg table {{ relation }}
+        target_lag = '{{ dynamic_table.target_lag }}'
+        warehouse = {{ dynamic_table.snowflake_warehouse }}
+        {{ optional('external_volume', catalog_relation.external_volume, "'") }}
+        catalog = 'SNOWFLAKE'  -- required, and always SNOWFLAKE for built-in Iceberg tables
+        base_location = '{{ catalog_relation.base_location }}'
+        {{ optional('refresh_mode', dynamic_table.refresh_mode) }}
+        {{ optional('initialize', dynamic_table.initialize) }}
+        {{ optional('row_access_policy', dynamic_table.row_access_policy) }}
```

Comment:
Is the syntax really different based on whether or not it's an Iceberg table?

Reply:
I don't believe it does vary by format/catalog but we still need it in the create statement right?

Reply:
also will see if we can get the diff a little cleaner here

Reply:
> I don't believe it does vary by format/catalog but we still need it in the create statement right?

This will render:
```sql
create dynamic iceberg table my_table
    ...
    row_access_policy = <row_access_user_config>
    table_tag = <table_tag_user_config>
    as (
        <sql>
    )
```
whereas the non-iceberg version above will render:
```sql
create dynamic iceberg table my_table
    ...
    with row access policy <row_access_user_config>
    with tag (<table_tag_user_config>)
    as (
        <sql>
    )
```
Note:
- `with` vs no `with`
- `=` vs no `=`
- `with` syntax uses `tag` instead of `table tag` or `table_tag`
- `with` syntax wraps the tags in `()`

Reply:
oh I misunderstood your point, will take a look